src/Microsoft.DotNet.Interactive.Formatting/IFormatContext.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System.IO;
+
+namespace Microsoft.DotNet.Interactive.Formatting
+{
+    public interface IFormatContext
+    {
+        /// <summary>Represents the approximate proportion of output width remaining to this format operation.</summary>
+        bool IsNestedInTable { get; }
```

Comment:
This seems very HTML-specific. I suspect there are other scenarios where the nesting of formatters might be of interest, e.g. is this potentially a generalization of recursion rules for plain text formatting or a way of thinking about creating drill-down points for visualizations that need to call back to the kernel?

Reply:
Yes. It's basically impossible not to either over-architect or under-architect this.  There are an infinite number of scenarios of interest, and each of them have an infinite number of possible refinements.  

One approach is make the context an immutable string --> object property bag and leave it up to different formatters to use whatever keys they like.   Another is to strongly type is according to our specific needs and then iterate/refine.  Switching to the immutable property bag (with a couple of predefined keys) at least ensures there's space for whatever information any formatter wants to register.

Reply:
I think that even the simple step of allowing the context to understand the current recursion depth and list expansion index could work. My assumption is that `IsNestedInTable` is currently functionally equivalent to `RecursionDepth > 1`.

Reply:
Random other ideas 

- If we want to keep this strongly typed, non-extensible and simple, then a more declarative name might be along the lines `SimpleDisplayRequested` indicating that some outer formatting has requested that items use a simpler display. A simple flag like this is easy enough to understand and is enough to allow all sorts of restrictions to kick in (reduce number of properties being displayed by default, no nested tables etc.)

- A slight variation in this iso to make that a number `ContentThreshold` starting at 1.0 for "full display" and reducing down in inner table recursions.

I don't really mind - maybe we could start with this version and move to a property bag later if needed.  Making IFormatContext a sealed immutable type would allow us to do whatever we liked in the long run (e.g. add more properties or a proprety bag).

I think I'll change to ContentThreshold for now.